### PR TITLE
Add a button to show new entities

### PR DESCRIPTION
**Pull request by c727**
to-do: fix dialog. set entityId to clipboard

```yaml
excluded_entities:
  - weblink.router
```

the dialog need some more tweaks. also not sure if this is the correct place to add it

### CODE REVIEW

File: src/panels/lovelace/common/compute-unused-entities.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  function addEntities(obj) {
+    if (obj.entity) addEntityId(obj.entity);
+    if (obj.entities) obj.entities.forEach(entity => addEntityId(entity));
+    if (obj.card) addEntities(obj.card);
+    if (obj.cards) obj.cards.forEach(card => addEntities(card));
+  }
+
+  config.views.forEach(view => addEntities(view));
+  return Array.from(entities);
```
Comment (balloob) [inline src/panels/lovelace/common/compute-unused-entities.js]: Why should we return it as an array? Sets are fast to check against (`O(1)`)

File: src/panels/lovelace/hui-root.js [hunk 2]
```diff
@@ -158,24 +174,28 @@ class HUIRoot extends NavigateMixin(EventsMixin(PolymerElement)) {
...
     }
 
-    const viewConfig = this.config.views[this._curView];
-
     let view;
 
-    if (viewConfig.panel) {
-      view = createCardElement(viewConfig.cards[0]);
+    if (viewIndex === 'unused') {
+      this._curView = -1;
```
Comment (balloob) [inline src/panels/lovelace/hui-root.js]: Why not just change it to unused and keep that assignment at top of the file? That keeps the code a little easier to read.

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
Looks great. Comments are optional 👍

**Comment by c727** [comment]
the menu does not close if I press the same item again...

**Comment by balloob** [comment]
It will close the menu if `iron-select` is fired. The trick is to reset the selection on the paper-listbox by setting `el.selected = null;`

**Pull request merged**